controllers/cbcontainersagent_controller.go

```diff
@@ -138,6 +147,17 @@ func (r *CBContainersAgentController) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	r.Log.Info("Finished reconciling", "Requiring", stateWasChanged)
+
+	if err = r.updateCRStatus(ctx, cbContainersAgent, stateWasChanged); err != nil {
+		if k8sErrors.IsConflict(err) {
```

Comment:
This is due to K8S' optimistic concurrency and is the recommended approach by`controller-runtime` (run another reconcile in case of conflict to get "fresh" data). 
`Getting` the resource would not work as it'd read from cache and still have the stale version.